bindings/cpp/src/android.cc

```diff
@@ -0,0 +1,101 @@
+#include <jni.h>
+#include <string>
+#include "defi-wallet-core-cpp/include/android.h"
+#define SECURE_STORAGE_CLASS "com/cronos/play/SecureStorage"
+using namespace std;
+
+JNIEnv *g_env = NULL;
+namespace org {
+namespace defi_wallet_core {
+
+int secureStorageSetJavaEnv(JNIEnv *userenv) {
+  g_env = userenv;
+  return 1;
+}
+
+int secureStorageWriteBasic(JNIEnv *env, string userkey, string uservalue) {
+
+  string secureStorageClass = SECURE_STORAGE_CLASS;
+  jclass activityThreadClass = env->FindClass("android/app/ActivityThread");
+  jmethodID currentActivityThreadMethod =
+      env->GetStaticMethodID(activityThreadClass, "currentActivityThread",
+                             "()Landroid/app/ActivityThread;");
+  jobject activityThread = env->CallStaticObjectMethod(
+      activityThreadClass, currentActivityThreadMethod);
+  jmethodID getApplicationMethod = env->GetMethodID(
+      activityThreadClass, "getApplication", "()Landroid/app/Application;");
+  jobject context = env->CallObjectMethod(activityThread, getApplicationMethod);
+  jclass kotlinClass = env->FindClass(secureStorageClass.c_str());
+  jmethodID functionMethod = env->GetStaticMethodID(
+      kotlinClass, "writeSecureStorage",
+      "(Landroid/content/Context;Ljava/lang/String;Ljava/lang/String;)I");
+  jstring key = env->NewStringUTF(userkey.c_str());
+  jstring value = env->NewStringUTF(uservalue.c_str());
+  jint ret = env->CallStaticIntMethod(kotlinClass, functionMethod, context, key,
+                                      value);
+
+  return (int)ret;
+}
+
+string secureStorageReadBasic(JNIEnv *env, string userkey) {
+
+  string secureStorageClass = SECURE_STORAGE_CLASS;
+  jclass activityThreadClass = env->FindClass("android/app/ActivityThread");
+  jmethodID currentActivityThreadMethod =
+      env->GetStaticMethodID(activityThreadClass, "currentActivityThread",
+                             "()Landroid/app/ActivityThread;");
+  jobject activityThread = env->CallStaticObjectMethod(
+      activityThreadClass, currentActivityThreadMethod);
+  jmethodID getApplicationMethod = env->GetMethodID(
+      activityThreadClass, "getApplication", "()Landroid/app/Application;");
+  jobject context = env->CallObjectMethod(activityThread, getApplicationMethod);
+  jclass kotlinClass = env->FindClass(secureStorageClass.c_str());
+  jmethodID functionMethod = env->GetStaticMethodID(
+      kotlinClass, "readSecureStorage",
+      "(Landroid/content/Context;Ljava/lang/String;)Ljava/util/HashMap;");
+
+  jstring x = env->NewStringUTF(userkey.c_str());
+  jobject ret =
+      env->CallStaticObjectMethod(kotlinClass, functionMethod, context, x);
+
+  jstring resultkey = env->NewStringUTF("result");
+  jstring successkey = env->NewStringUTF("success");
+  jstring errorkey = env->NewStringUTF("error");
+  jclass mapClass = env->FindClass("java/util/HashMap");
```

Comment:
I guess one alternative is to define a Kotlin/Java data class with those three fields?

Reply:
yes, that's possible, i'll check

Reply:
return json directly